src/applications/PerfusionDerivatives.h

```diff
@@ -102,40 +100,30 @@ class PerfusionDerivatives
   std::vector<double> CalculateAveragePerfusionCurve(typename PerfusionImageType::Pointer perfImagePointerNifti);
 
   template<class ImageType, class PerfusionImageType >
-  bool CheckPerfusionQuality(typename PerfusionImageType::Pointer perfImagePointerNifti, const std::string outputdirectory);
+  bool IsPerfusionQualityGood(typename PerfusionImageType::Pointer perfImagePointerNifti, const std::string outputdirectory);
 };
 
 template< class ImageType, class PerfusionImageType >
 std::vector<double> PerfusionDerivatives::CalculateAveragePerfusionCurve(typename PerfusionImageType::Pointer perfImagePointerNifti)
 {
-  std::vector<double> AverageCurve;
-  ImageTypeFloat4D::RegionType region = perfImagePointerNifti->GetLargestPossibleRegion();
-  for (unsigned int volumes = 0; volumes < region.GetSize()[3]; volumes++)
+  PerfusionImageType::RegionType region = perfImagePointerNifti->GetLargestPossibleRegion();
```

Comment:
I think you need a "typename" before "PerfusionImageType::RegionType" here.

Reply:
added typename